File: CS/PivotGridDemo.Wpf/Helpers/ServerMode/ServerModeViewModel.cs
using DevExpress.Xpf.DemoBase;
using PivotGridDemo.Helpers;
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Linq;
using System.Text;

namespace PivotGridDemo {
    public class ServerModeViewModel : ServerModeViewModelBase<OrderDataRecord> {
        #region DatabaseInfo
        public static readonly DatabaseInfo DatabaseInfo = new DatabaseInfo(
            OrderDataContext.FileName, "OrderDataRecords", typeof(OrderDataRecord), CreateValues, (sql, connection) => new SQLiteCommand(sql, (SQLiteConnection)connection));
        static Dictionary<string, object> CreateValues() {
            var dict = new Dictionary<string, object>();
            var product = DatabaseHelper.GetProduct();
            dict.Add("OrderDate", DatabaseHelper.GetOrderDate());
            dict.Add("Quantity", DatabaseHelper.GetQuantity());
            dict.Add("UnitPrice", DatabaseHelper.GetProductPrice(product));
            dict.Add("CustomerName", DatabaseHelper.GetCustomerName());
            dict.Add("ProductName", product.ProductName);
            dict.Add("CategoryName", product.CategoryName);
            dict.Add("SalesPersonName", DatabaseHelper.GetSalesPersonName());
            return dict;
        }
        #endregion

        protected ServerModeViewModel()
            : base(DatabaseInfo, () => new OrderDataContext().OrderDataRecords) {
        }
    }
}

File: VB/PivotGridDemo.Wpf/Helpers/ServerMode/ServerModeViewModel.vb
Imports DevExpress.Xpf.DemoBase
Imports PivotGridDemo.Helpers
Imports System
Imports System.Collections.Generic
Imports System.Data.SQLite
Imports System.Linq
Imports System.Text

Namespace PivotGridDemo
    Public Class ServerModeViewModel
        Inherits ServerModeViewModelBase(Of OrderDataRecord)

        #Region "DatabaseInfo"
        Public Shared ReadOnly DatabaseInfo As New DatabaseInfo(OrderDataContext.FileName, "OrderDataRecords", GetType(OrderDataRecord), AddressOf CreateValues, Function(sql, connection) New SQLiteCommand(sql, CType(connection, SQLiteConnection)))
        Private Shared Function CreateValues() As Dictionary(Of String, Object)
            Dim dict = New Dictionary(Of String, Object)()
            Dim product = DatabaseHelper.GetProduct()
            dict.Add("OrderDate", DatabaseHelper.GetOrderDate())
            dict.Add("Quantity", DatabaseHelper.GetQuantity())
            dict.Add("UnitPrice", DatabaseHelper.GetProductPrice(product))
            dict.Add("CustomerName", DatabaseHelper.GetCustomerName())
            dict.Add("ProductName", product.ProductName)
            dict.Add("CategoryName", product.CategoryName)
            dict.Add("SalesPersonName", DatabaseHelper.GetSalesPersonName())
            Return dict
        End Function
        #End Region

        Protected Sub New()
            MyBase.New(DatabaseInfo, Function() (New OrderDataContext()).OrderDataRecords)
        End Sub
    End Class
End Namespace
